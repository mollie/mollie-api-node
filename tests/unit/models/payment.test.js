import Payment from 'models/payment';

describe('payment model', () => {
  it('should instantiate with default values', () => {
    const payment = new Payment();

    expect(payment.isOpen()).toBe(false);
    expect(payment.isPaid()).toBe(false);
    expect(payment.isCanceled()).toBe(false);
    expect(payment.isExpired()).toBe(false);
    expect(payment.getPaymentUrl()).toBeNull();
    expect(payment.toPlainObject()).toMatchSnapshot();
  });

  it('should instantiate with given values', () => {
    const paymentProps = JSON.parse(`{
        "resource": "payment",
        "id": "tr_7UhSN1zuXS",
        "mode": "test",
        "createdAt": "2018-03-20T09:13:37+00:00",
        "amount": {
            "value": "10.00",
            "currency": "EUR"
        },
        "description": "My first payment",
        "method": null,
        "metadata": {
            "order_id": "12345"
        },
        "status": "open",
        "isCancelable": false,
        "expiresAt": "2018-03-20T09:28:37+00:00",
        "details": null,
        "profileId": "pfl_QkEhN94Ba",
        "sequenceType": "oneoff",
        "redirectUrl": "https://webshop.example.org/order/12345/",
        "webhookUrl": "https://webshop.example.org/payments/webhook/",
        "_links": {
            "self": {
                "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
                "type": "application/json"
            },
            "checkout": {
                "href": "https://www.mollie.com/payscreen/select-method/7UhSN1zuXS",
                "type": "text/html"
            },
            "documentation": {
                "href": "https://docs.mollie.com/reference/v2/payments-api/create-payment",
                "type": "text/html"
            }
        }
    }`);
    const payment = new Payment(paymentProps);

    expect(payment.isOpen()).toBe(true);
    expect(payment.isPaid()).toBe(false);
    expect(payment.isCanceled()).toBe(false);
    expect(payment.isExpired()).toBe(false);
    expect(payment.getPaymentUrl()).toBe(paymentProps._links.checkout.href);
    expect(payment.toPlainObject()).toMatchSnapshot();
  });
});
