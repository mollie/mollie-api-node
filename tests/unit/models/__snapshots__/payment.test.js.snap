// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`payment model should instantiate with default values 1`] = `
Object {
  "_links": Object {
    "chargebacks": null,
    "checkout": null,
    "customer": null,
    "mandate": null,
    "refunds": null,
    "settlement": null,
    "subscription": null,
  },
  "amount": Object {
    "currency": null,
    "value": null,
  },
  "amountRefunded": null,
  "amountRemaining": null,
  "applicationFee": Object {
    "amount": Object {
      "currency": null,
      "value": null,
    },
    "description": null,
  },
  "canceledAt": null,
  "countryCode": null,
  "createdAt": null,
  "customerId": null,
  "description": null,
  "details": null,
  "expiredAt": null,
  "expiresAt": null,
  "failedAt": null,
  "id": null,
  "isCancelable": null,
  "locale": null,
  "mandateId": null,
  "metadata": null,
  "method": null,
  "mode": null,
  "paidAt": null,
  "profileId": null,
  "redirectUrl": null,
  "resource": "payment",
  "sequenceType": null,
  "settlementAmount": null,
  "settlementId": null,
  "status": null,
  "subscriptionId": null,
  "webhookUrl": null,
}
`;

exports[`payment model should instantiate with given values 1`] = `
Object {
  "_links": Object {
    "checkout": Object {
      "href": "https://www.mollie.com/payscreen/select-method/7UhSN1zuXS",
      "type": "text/html",
    },
    "documentation": Object {
      "href": "https://docs.mollie.com/reference/v2/payments-api/create-payment",
      "type": "text/html",
    },
    "self": Object {
      "href": "https://api.mollie.com/v2/payments/tr_7UhSN1zuXS",
      "type": "application/json",
    },
  },
  "amount": Object {
    "currency": "EUR",
    "value": "10.00",
  },
  "amountRefunded": null,
  "amountRemaining": null,
  "applicationFee": Object {
    "amount": Object {
      "currency": null,
      "value": null,
    },
    "description": null,
  },
  "canceledAt": null,
  "countryCode": null,
  "createdAt": "2018-03-20T09:13:37+00:00",
  "customerId": null,
  "description": "My first payment",
  "details": null,
  "expiredAt": null,
  "expiresAt": "2018-03-20T09:28:37+00:00",
  "failedAt": null,
  "id": "tr_7UhSN1zuXS",
  "isCancelable": false,
  "locale": null,
  "mandateId": null,
  "metadata": Object {
    "order_id": "12345",
  },
  "method": null,
  "mode": "test",
  "paidAt": null,
  "profileId": "pfl_QkEhN94Ba",
  "redirectUrl": "https://webshop.example.org/order/12345/",
  "resource": "payment",
  "sequenceType": "oneoff",
  "settlementAmount": null,
  "settlementId": null,
  "status": "open",
  "subscriptionId": null,
  "webhookUrl": "https://webshop.example.org/payments/webhook/",
}
`;
